extensions/wikia/PortableInfobox/services/PortableInfoboxRenderService.class.php

```diff
@@ -263,15 +263,25 @@ private function createSmartGroups( $groupData, $rowCapacity ) {
 			}
 		}
 
-		return array_merge( $result, $this->applyRowItemsStyles( $rowItems, $rowSpan ) );
+		return array_merge( $result, $this->applyRowItemsStyles( $rowItems, $rowSpan, true ) );
 	}
 
-	private function applyRowItemsStyles( $rowItems, $capacity ) {
-		return array_map( function ( $item ) use ( $capacity ) {
+	private function applyRowItemsStyles( $rowItems, $capacity, $isLastRow = false ) {
+		$items = array_map( function ( $item, $index ) use ( $capacity, $isLastRow ) {
 			$width = round( ( $item['data']['span'] / $capacity ) * 100 );
 			$item['data']['cssClasses'] = 'pi-smart-data';
+			if ( $isLastRow ) {
+				$item['data']['cssClasses'] .= ' last-row';
+			}
+			if ( $index % 2 === 1 ) {
+				$item['data']['cssClasses'] .= ' even-in-row';
```

Comment:
Let's keep those names as short as possible, but please keep the `pi-` prefix. As for the BEM notation, it is not used in PIs anywhere, so let's be consistent with that, and remember that users use those class names as well.